docs/how-to-build.md

```diff
@@ -4,244 +4,232 @@ slug: /how-to-build
 
-1. Clone the Gravitino project.
-
-   If you want to contribute to this open-source project, please fork the project on GitHub first. After forking, clone the forked project to your local environment, make your changes, and submit a pull request (PR).
+:::note
+Depending on how you deploy Gravitino, there may be contain known security vulnerabilities
+in other software used in conjunction with Gravitino.
+:::
 
-   ```shell
-   git clone git@github.com:apache/gravitino.git
-   ```
+### Install WSL on Windows
 
-2. Build the Gravitino project. Running this for the first time can take 15 minutes or more.
+If you are building Gravitino on a Linux or macOS machine, you can proceed to the [next section](#install-jdk).
+To build Apache Gravitino on Windows, you need to install the Windows Subsystem for Linux (WSL)
+from Microsoft which provides you a Linux-like environment for development.
+For installing WSL, refer to the [WSL Installation Guide](https://learn.microsoft.com/en-us/windows/wsl/install).
+The following steps assume that you have installed Unbuntu which is the default distribution for WSL.
 
-   ```shell
-   cd gravitino
-   ./gradlew build
-   ```
+*Note: Gravitino can run successfully on Ubuntu 22.04.*
 
-   The default specified JDK version is 8, but if you want to use JDK 11 or 17 to build, modify the property `jdkVersion` to 11 or 17 in the `gradle.properties` file, or specify the version with `-P`, like:
+Make sure you have fresh information about the latest versions of Ubuntu packages and their dependencies:
 
-   ```shell
-   ./gradlew build -PjdkVersion=11
-   ```
+```shell
+sudo apt update
+```
 
-   Or:
+Install the prerequisite packages required by later software installation:
 
-   ```shell
-   ./gradlew build -PjdkVersion=17
-   ```
+```shell
+sudo apt install apt-transport-https ca-certificates curl software-properties-common
+```
 
-  The `./gradlew build` command builds all the Gravitino components, including the Gravitino server, Java and Python clients, Trino and Spark connectors, and more.
+### Install JDK
 
-  For the Python client, the `./gradlew build` command builds the Python client with Python 3.8 by default. If you want to use Python 3.9, 3.10, or 3.11 to build, please modify the property `pythonVersion` to 3.9, 3.10, or 3.11 in the `gradle.properties` file, or specify the version with `-P` like:
+You don't have to preinstall the JDK environment.
+The Gradle Java toolchain can detect and install JDK automatically by running the `./gradlew javaToolchains` command.
+For more details, see [Gradle Java Toolchain](https://docs.gradle.org/current/userguide/toolchains.html#sec:java_toolchain).
 
-   ```shell
-   ./gradlew build -PpythonVersion=3.9
-   ```
+If you want to install JDK manually, Java SDK 17 is recommended, and JDK 8 or JDK 11 also works.
+After installing the JDK, you may want to setup your Shell environment ensure that the correct JDK version will be used.
```

Comment:
"set up your Shell environment to ensure"

Reply:
ok.